### DISCUSSION TITLE
Recommended approach of Linq2DB implementation in Blazor Server Application

### DISCUSSION THREAD

**Discussion by plachow** [linq2db/linq2db#3947] [Q&A] [answered]
Hi, what is recommended approach to use Linq2DB in Blazor Server environment? Its scope is different from other typical web apps. It exists much longer and I want to leave database connection open as short time as possible.
1) Use dependency injection? It works but I'm notsure about Blazor Server and its "AddScoped". Scope will remain throug whole active "circuit". This can be also hours.

```
builder.Services.AddScoped<IFileService, FilesService>();
builder.Services.AddLinqToDBContext<Ldc>((provider, options) =>
{
    options   
        .UseSqlServer(builder.Configuration.GetConnectionString("Default"))
        .UseDefaultLogging(provider);
});
```

then inject context to class and consume it:

```
public class FilesService : IFileService
    {

        private readonly Ldc dc;
        public FilesService(Ldc connection)
        {
            dc = connection;
        }


        public async Task<int> Count()
        {
            return await dc.Files.CountAsync();
        }
}
```

2) Use old fashioned approach with creating context on demand and query data inside "using":
`public Ldc() : base(ProviderName.SqlServer2016, "CONNECTION_STRING") { }`

and

```
await using Ldc dc=new();
List<Product>products=await dc.Products.ToListAsync();

```
Or exists some other recommended approach? 

Thank you and keep your work rolling. Linq2DB is the far best ORM!

**Comment by igor-tkachev** [comment, marked as answer]
2. As implicit control leads to more unwanted surprises.
